### PR TITLE
cut-out train step [wip]

### LINKED ISSUE #708: LR Schedulers shouldn't get `epoch` argument in `step` function
## 🐛 Bug

PyTorch LR schedulers now shouldn't get any arguments in `step` function, see [here](https://github.com/pytorch/pytorch/blob/9e9bfbfd8d893cdf8205db67f220551214787d7f/torch/optim/lr_scheduler.py#L13-L20) and [here](https://github.com/pytorch/pytorch/issues/31828).

Looks like the calls in PytorchLightning are not in line with the new interface, see [here](https://github.com/PyTorchLightning/pytorch-lightning/blob/dac59bb8d354f28b919d08aa369a7db6ab31bfa6/pytorch_lightning/trainer/training_loop.py#L334-L337).

This results in unexpected LR changes. Removing the `epoch` argument from `step` call solves the issue for me.

## Environment
PyTorch 1.4
PyTorchLightning 0.5.3.2

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?
cut out too large train epoch into a separate method, see https://github.com/PyTorchLightning/pytorch-lightning/pull/856#discussion_r379860319

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

**Comment by Borda** [comment]
Tests is failing because of some cycling...

File: pytorch_lightning/trainer/training_loop.py [hunk 1]
```diff
@@ -304,86 +297,92 @@ def process_output(self, output, train):
...
+                #  see the base class `torch.optim.lr_scheduler._LRScheduler`
+                lr_scheduler.last_epoch = self.current_epoch
+        if self.reduce_lr_on_plateau_scheduler is not None:
+            val_loss = self.callback_metrics.get('val_loss')
+            if val_loss is None:
+                avail_metrics = ','.join(list(self.callback_metrics.keys()))
+                m = "ReduceLROnPlateau conditioned on metric val_loss" \
+                    f" which is not available. Available metrics are: {avail_metrics}"
+                raise MisconfigurationException(m)
+            self.reduce_lr_on_plateau_scheduler.step(val_loss, epoch=self.current_epoch)
```
Comment (ethanwharris) [inline pytorch_lightning/trainer/training_loop.py]: Why use `.step` here if not above?

File: pytorch_lightning/trainer/training_loop.py [hunk 2]
```diff
@@ -304,86 +297,92 @@ def process_output(self, output, train):
...
+        # RUN TRAINING EPOCH
+        # -----------------
+        self.run_training_epoch()
+
+        # update LR schedulers
+        if self.lr_schedulers:
+            for lr_scheduler in self.lr_schedulers:
+                # accessing the Scheduler state directly,
+                #  see the base class `torch.optim.lr_scheduler._LRScheduler`
+                lr_scheduler.last_epoch = self.current_epoch
```
Comment (ethanwharris) [inline pytorch_lightning/trainer/training_loop.py]: Not sure this is a good idea, any alternatives that involve just calling `.step`?

Comment (Borda) [reply pytorch_lightning/trainer/training_loop.py]: well I can make a loop over and call `.step()` until the `.last_epoch` is the `current_epoch - 1`

File: pytorch_lightning/trainer/training_loop.py (hunk 1 above)
Comment (Borda) [reply pytorch_lightning/trainer/training_loop.py]: thx, I have overlooked this one :]

**Comment by Borda** [comment]
unless set differently at the init the, it starts steps from 0, on the other hand, there is a desired feature to resume the process from step N so you need to set the particular step... but I see your point that it would be better to do this at scheduler creation...

**Comment by ethanwharris** [comment]
Yeah, it's a difficult choice but the while loop solution seems good. At least this way if something important happens in `.step` then we are covered :)

**Comment by Borda** [comment]
the original issue with schedulers was solved by #890
